rgw/rgw.go

```diff
@@ -0,0 +1,805 @@
+package rgw
```

Comment:
I strongly recommend splitting this file up into multiple smaller files. It makes the code simpler to understand and clearer what C functions are used by what.